### PR TITLE
Prior to this change, server error responses would not properly resol…

### PR DESCRIPTION
**Pull request by blittle**
…ve to request handlers

### CODE REVIEW

File: src/HTTPBackend.js [hunk 1]
```diff
@@ -30,20 +30,46 @@ export default function(options) {
...
 		return new Promise(function(resolve, reject) {
-			response.json().then((json) => {
-
-				let headers = { ...request.header };
+			let headers = { ...request.header };
+			delete headers.resource;
+			delete headers.parameters;
+			delete headers.queryParameters;
+
+			response.text().then(text => {
```
Comment (joeldenning) [inline src/HTTPBackend.js]: I usually wrap `.text()` or `.json()` in an if statement like this:

```js
if (response.ok) {
  response.text()...
} else {
  reject("Server responded with status " + response.status);
}
```

Not sure if there's really much of a difference, but just thought I'd throw it out there.

File: src/HTTPBackend.js [hunk 2]
```diff
@@ -30,20 +30,46 @@ export default function(options) {
...
+				try {
+					const json = JSON.parse(text);
+					resolve({
+						header: { ...headers, statusCode: response.status },
+						body: json
+					});
+				} catch (err) {
+					resolve({
+						header: { ...headers, statusCode: response.status },
```
Comment (blittle) [inline src/HTTPBackend.js]: There is an interesting case here where the response _might_ be a 200, yet have invalid json. So rxws would route the text response as a non-error instead.

**Comment by joeldenning** [comment]
lgtm, the one comment is not a biggy.

🚶

File: src/HTTPBackend.js (hunk 1 above)
Comment (blittle) [inline src/HTTPBackend.js]: I like that, though I will probably still potentially want the error message to resolve back to the original requester

**Pull request merged**